reframe/core/config.py

```diff
@@ -217,3 +220,151 @@ def create_env(system, partition, name):
                 system.add_partition(part)
 
             self._systems[sys_name] = system
+
+
+def ppretty(value, htchar=' ', lfchar='\n', indent=4, current_depth=0):
+    nlch = lfchar + htchar * indent * (current_depth + 1)
+    if type(value) is dict:
+        if value == {}:
+            return '{}'
+        items = [
+            nlch + repr(key) + ': ' +
+            ppretty(value[key], htchar, lfchar, indent, current_depth + 1)
+            for key in value
+        ]
+        return '{%s}' % (','.join(items) + lfchar +
```

Comment:
If you decide using `repr` and not `str`, you should use `%r` in format strings.

```suggestion
        return '{%r}' % (','.join(items) + lfchar +
```